### ISSUE TITLE
The glyph->has_outline method (and ensure_outline_loaded) should not croak

### ISSUE THREAD

**Issue by HinTak** [zmughal/p5-Font-FreeType#25]
It should just default to false. The main difference is with the .notdef glyph in an outline font . It still hasn't got any outlines but that's expected and not a fatal error. At the moment trying to load .notdef causes a fatal error. That's not nice.

You can try it out with adding $_->has_outline to list-characters vs list-glyphs, on the Vera font.

**Comment by HinTak** [comment]
I think some of the errchk() elsewhere should not be fatal either.

**Comment by HinTak** [comment]
I think has_outline itself should be strictly true/false; the other usage of ensure_outline_loaded may or may not need to croak on .notdef . I think .notdef should be false rather than croak.

**Comment by HinTak** [comment]
How do you feel about changing the code like this:

https://github.com/HinTak/p5-Font-FreeType/commit/dca5387508b07d84e2682304d5aca1771b3018a9
'is_outline() should default to false if anything goes wrong? '

**Comment by HinTak** [comment]
and please feel free to cherry-pick that commit from that branch (or I can issue a pull on a different new branch if you prefer).

**Comment by basiliscos** [comment]
> 'is_outline() should default to false if anything goes wrong? '
I don't see any reason for "lowering" interface to C-like API. 

The original author already chosen exception-mechanism, and we should use it consistently. Somewhere returning false and somewhere throwing an exception does not look good.

Also, this is breaking change. PPL, using the library in their apps, might complain. Give me some strong reason for that change?

**Comment by HinTak** [comment]
Fair enough. As mentioned in my latest post in my pull, my interests in this is wrapping around the font analysis backend of FontVal (which is a customized/enhanced version of FreeType), and as such, inputs of buggy fonts or less-than-perfect fonts is perhaps the norm. So it is important to keep going even when encountering problems, because problems *are* expected.

I just feel it is a bit confusing and surprising for a routine which supposedly returns true/false can throw an exception. I am okay wrapping stuff with eval on the client side, so I don't really mind this is rejected.

There are also instances upstream where FreeType tries to cope (silently) on buggy fonts, and that's important for system-wide rendering to not ever crash.
